sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -3393,9 +3400,16 @@ pi_result cuda_piEnqueueMemUnmap(pi_queue command_queue, pi_mem memobj,
       retEvent);
   } else {
     if (retEvent) {
-      *retEvent = _pi_event::make_native(PI_COMMAND_TYPE_MEM_BUFFER_UNMAP,
-                                         command_queue);
-      (*retEvent)->record();
+      try {
+        assert(command_queue);
```

Comment:
Suggest moving it to the begging of the function(~line 3391).

Reply:
This has also been moved to the start of its function.